frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -1569,31 +1569,29 @@ test("typeahead_results", () => {
     assert_mentions_matches("delia lear", []);
     assert_mentions_matches("Mark Tw", [twin1, twin2]);
 
+    // Earlier user group and stream mentions were autocompleted by their
+    // description too. This is now removed as it often led to unexpected
+    // behaviour, and did not have any great discoverability advantage.
+
     // Autocomplete user group mentions by group name.
     assert_mentions_matches("hamletchar", [hamletcharacters]);
 
-    // Autocomplete user group mentions by group descriptions.
-    assert_mentions_matches("characters ", [hamletcharacters]);
-    assert_mentions_matches("characters of ", [hamletcharacters]);
-    assert_mentions_matches("characters o ", []);
-    assert_mentions_matches("haracters of hamlet", []);
-    assert_mentions_matches("of hamlet", [hamletcharacters]);
-
     // Autocomplete by slash commands.
     assert_slash_matches("me", [me_slash]);
     assert_slash_matches("dark", [dark_slash]);
     assert_slash_matches("night", [dark_slash]);
     assert_slash_matches("light", [light_slash]);
     assert_slash_matches("day", [light_slash]);
 
-    // Autocomplete stream by stream name or stream description.
+    // Autocomplete stream by stream name
     assert_stream_matches("den", [denmark_stream, sweden_stream]);
     assert_stream_matches("denmark", [denmark_stream]);
     assert_stream_matches("denmark ", []);
     assert_stream_matches("den ", []);
-    assert_stream_matches("cold", [sweden_stream, denmark_stream]);
     assert_stream_matches("the ", [netherland_stream]);
-    assert_stream_matches("city", [netherland_stream]);
+
+    assert_stream_matches("cold", []);
```

Comment:
Perhaps a comment like `// Do not match stream descriptions` would help with the readability here.